app/services/user.py

```diff
@@ -139,6 +139,26 @@ def update(self) -> User:
         return self.user
 
 
+class UserDeletionService:
+    def __init__(self, user, **attrs):
+        self.user = user
+        self.scopes = attrs.pop("scopes")
+        authentication.validate_scope_permissions("delete:user", self.scopes, attrs)
+        self.attrs = attrs
```

Comment:
No attrs are ever passed. Also we need to delete all tunnels and subdomains associated with the user

Reply:
The scopes are being passed in the `attrs` parameter